### PR TITLE
Split jointags into jointags and pivottags

### PR DESCRIPTION
**Pull request by vertexmc**
- (breaking change) jointags loses the `keep_nodes` kwarg
- (breaking change) jointags no longer acts as a pivot by default
- adds pivottags operator

### CODE REVIEW

File: synapse/tests/test_lib_storm.py [hunk 1]
```diff
@@ -527,27 +527,86 @@ def test_storm_tag_jointag(self):
...
+            self.eq(nodes[1][1].get('tufo:form'), 'inet:fqdn')
+            nodes = core.eval('inet:dns:a pivottags(inet:ipv4,inet:dns:a)')  # there is no ipv4
+            self.len(1, nodes)
+            self.eq(nodes[0][1].get('tufo:form'), 'inet:dns:a')
+            nodes = core.eval('inet:dns:a pivottags(inet:ipv4)')  # there is no ipv4
+            self.len(0, nodes)
+
+            # Demonstrate that limits work
+            self.len(2, core.eval('inet:dns:a pivottags(limit=3)'))
+            self.len(2, core.eval('inet:dns:a pivottags(limit=2)'))  # sometimes is broken
```
Comment (vertexmc) [inline synapse/tests/test_lib_storm.py]: This branch is still broken. I get different results on these queries on the same data.

Comment (vertexmc) [reply synapse/tests/test_lib_storm.py]: Of the four queries I run here, the only one that seems to get occasionally different/incorrect results is `inet:dns:a pivottags(limit=2)`

File: docs/synapse/userguides/ug015_storm_ref_pivot.rst [hunk 2]
```diff
@@ -335,7 +335,8 @@ N/A
 
 jointags()
 ----------
-Returns all specified nodes that have **any** of the tags applied to **any** of the working set of nodes.
+Returns all specified nodes that have **any** of the tags applied to **any** of the working set of nodes,
```
Comment (therealsilence) [inline docs/synapse/userguides/ug015_storm_ref_pivot.rst]: Nitpicky change suggestion: "Returns the current working set **plus** all nodes that have **any** of the tags applied to **any** node in the working set."

File: docs/synapse/userguides/ug015_storm_ref_pivot.rst [hunk 3]
```diff
@@ -375,9 +376,60 @@ N/A
 
 **Usage notes:**
 
-* ``jointags()`` pivots using the set of leaf tags only. For example if nodes in the working set have the tag ``#foo.bar.baz``, ``jointags()`` will return other nodes with ``#foo.bar.baz``, but not nodes with ``#foo.bar`` or ``#foo`` alone.
-* ``jointags()``, like ``refs()``, can be useful to "explore" other nodes that share some analytical assessment (tag) with the working set of nodes, but may return a large number of nodes. It may be more efficient to narrow the scope of the query using ``totags()`` in combination with a filter operator (e.g., to limit the specific tags selected) followed by ``fromtags()``.
-* The ``limit=`` parameter can be provided as input to the ``jointags()`` operator itself when using Operator syntax. Alternately the ``limit()`` operator_ can be used after the ``jointags()`` operator (in either Operator or Macro syntax) to specify a limit on the number of nodes returned.
+* ``jointags()`` joins using the set of leaf tags only. For example if nodes in the working set have the tag ``#foo.bar.baz``, ``jointags()`` will return other nodes with ``#foo.bar.baz``, but not nodes with ``#foo.bar`` or ``#foo`` alone.
```
Comment (therealsilence) [inline docs/synapse/userguides/ug015_storm_ref_pivot.rst]: Suggest adding bullet point at top of usage notes list:
- jointags() does **not** consume nodes by design and so includes (retains) the original working set as part of the resulting set.

File: docs/synapse/userguides/ug015_storm_ref_pivot.rst [hunk 4]
```diff
@@ -375,9 +376,60 @@ N/A
...
+* ``jointags()``, like ``refs()``, can be useful to "explore" other nodes that share some analytical assessment (tag) with the working set of nodes, but may return a large number of nodes.
+It may be more efficient to narrow the scope of the query using ``totags()`` in combination with a filter operator (e.g., to limit the specific tags selected) followed by ``fromtags()``.
+* The ``limit=`` parameter can be provided as input to the ``jointags()`` operator itself when using Operator syntax.
+Alternately the ``limit()`` operator_ can be used after the ``jointags()`` operator (in either Operator or Macro syntax) to specify a limit on the number of nodes returned.
+
+pivottags()
+----------
+Returns all specified nodes that have **any** of the tags applied to **any** of the working set of nodes,
```
Comment (therealsilence) [inline docs/synapse/userguides/ug015_storm_ref_pivot.rst]: Nitpicky change suggestion: "Returns all nodes that have **any** of the tags applied to **any** node in the original working set; excludes the original working set from the results."

File: docs/synapse/userguides/ug015_storm_ref_pivot.rst [hunk 5]
```diff
@@ -375,9 +376,60 @@ N/A
...
+removing all other data from the query results.
+
+``pivottags()`` can be thought of as executing a ``totags()`` operation followed by a ``fromtags()`` operation.
```
Comment (therealsilence) [inline docs/synapse/userguides/ug015_storm_ref_pivot.rst]: I would remove this statement as it's not technically true for pivottags(), only for jointags() (pivoting totags and then fromtags will include ALL nodes with the relevant tags, where pivottags excludes the original nodes.

File: docs/synapse/userguides/ug015_storm_ref_pivot.rst [hunk 6]
```diff
@@ -375,9 +376,60 @@ N/A
...
+* ``jointags()`` joins using the set of leaf tags only. For example if nodes in the working set have the tag ``#foo.bar.baz``, ``jointags()`` will return other nodes with ``#foo.bar.baz``, but not nodes with ``#foo.bar`` or ``#foo`` alone.
+* ``jointags()``, like ``refs()``, can be useful to "explore" other nodes that share some analytical assessment (tag) with the working set of nodes, but may return a large number of nodes.
+It may be more efficient to narrow the scope of the query using ``totags()`` in combination with a filter operator (e.g., to limit the specific tags selected) followed by ``fromtags()``.
+* The ``limit=`` parameter can be provided as input to the ``jointags()`` operator itself when using Operator syntax.
+Alternately the ``limit()`` operator_ can be used after the ``jointags()`` operator (in either Operator or Macro syntax) to specify a limit on the number of nodes returned.
```
Comment (therealsilence) [inline docs/synapse/userguides/ug015_storm_ref_pivot.rst]: Should we clarify how "limit" works for jointags() vs pivottags()? e.g., for a join-type operation, "limit" represents the maximum number of nodes **added** to the results (e.g., the original working set, plus "limit" number of additional nodes)? Similar for pivottags() below (see).

File: docs/synapse/userguides/ug015_storm_ref_pivot.rst [hunk 7]
```diff
@@ -375,9 +376,60 @@ N/A
...
+* Return the set of ``inet:fqdn`` and ``inet:email`` nodes that share any of the tags applied to the working set of nodes, limiting the number of results to 10:
+  ::
+    pivottags( inet:fqdn, inet:email, limit=10 )
+
+**Usage notes:**
+
+* ``pivottags()`` pivots using the set of leaf tags only. For example if nodes in the working set have the tag ``#foo.bar.baz``, ``pivottags()`` will return other nodes with ``#foo.bar.baz``, but not nodes with ``#foo.bar`` or ``#foo`` alone.
+* ``pivottags()``, like ``refs()``, can be useful to "explore" other nodes that share some analytical assessment (tag) with the working set of nodes, but may return a large number of nodes.
+It may be more efficient to narrow the scope of the query using ``totags()`` in combination with a filter operator (e.g., to limit the specific tags selected) followed by ``fromtags()``.
+* The ``limit=`` parameter can be provided as input to the ``pivottags()`` operator itself when using Operator syntax.
```
Comment (therealsilence) [inline docs/synapse/userguides/ug015_storm_ref_pivot.rst]: Should we clarify how "limit" works for pivottags() vs pivottags()? e.g., for a pivot-type operation, "limit" represents the **total** number of nodes returned? Contrast with jointags() above (see).

File: docs/synapse/userguides/ug015_storm_ref_pivot.rst [hunk 8]
```diff
@@ -375,9 +376,60 @@ N/A
...
+* Return the set of ``inet:fqdn`` and ``inet:email`` nodes that share any of the tags applied to the working set of nodes:
+  ::
+    pivottags( inet:fqdn, inet:email )
+
+* Return the set of ``inet:fqdn`` and ``inet:email`` nodes that share any of the tags applied to the working set of nodes, limiting the number of results to 10:
+  ::
+    pivottags( inet:fqdn, inet:email, limit=10 )
+
+**Usage notes:**
+
```
Comment (therealsilence) [inline docs/synapse/userguides/ug015_storm_ref_pivot.rst]: Suggest adding bullet at top, "pivottags() consumes nodes input into the function, so the resulting set of nodes will **not** include nodes from the original working set."

**Pull request merged**